src/http_client/mod.rs

```diff
@@ -0,0 +1,73 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use reqwest::Url;
+
+pub struct HttpClientImpl {}
+
+#[derive(serde::Deserialize, serde::Serialize, Debug)]
+pub struct CompilerMeetings {
+    items: Vec<CompilerMeeting>,
+}
+
+#[derive(serde::Deserialize, serde::Serialize, Debug)]
+pub struct Start {
+    #[serde(rename(deserialize = "dateTime"))]
+    date_time: String,
+}
+
+#[derive(serde::Deserialize, serde::Serialize, Debug)]
+pub struct CompilerMeeting {
+    summary: String,
+    #[serde(rename(deserialize = "htmlLink"))]
+    html_link: String,
+    #[serde(rename(deserialize = "originalStartTime"))]
+    original_start: Option<Start>,
+    start: Option<Start>,
+}
+
+#[async_trait]
+pub trait HttpClient {
+    async fn get_meetings(
+        &self,
+        start_date: String,
+        end_date: String,
+    ) -> Result<Vec<CompilerMeeting>>
+    where
+        Self: Sized;
+}
+
+#[async_trait]
+impl HttpClient for HttpClientImpl {
+    /// Retrieve all meetings from the Rust Compiler Team Calendar in a date range
+    // Google calendar API documentation:
+    // https://developers.google.com/calendar/api/v3/reference/events/list
+    // The API token needs only one permission: https://www.googleapis.com/auth/calendar.events.readonly
+    async fn get_meetings(
+        &self,
+        start_date: String,
+        end_date: String,
+    ) -> Result<Vec<CompilerMeeting>> {
+        let api_key = match std::env::var("GOOGLE_API_KEY") {
+            Ok(v) => v,
+            Err(_) => panic!("Missing find GOOGLE_API_KEY in env vars"),
+        };
+        let google_calendar_id = "6u5rrtce6lrtv07pfi3damgjus%40group.calendar.google.com";
+        let time_min = format!("{}T09:00:00+02:00", start_date);
+        let time_max = format!("{}T23:00:00+02:00", end_date);
+        let url = Url::parse_with_params(
+            &format!(
+                "https://www.googleapis.com/calendar/v3/calendars/{}/events",
+                google_calendar_id
+            ),
+            &[
+                // see google docs for the meaning of these values
```

Comment:
Maybe add a link to the docs in the comments?

Reply:
I've put a link to the docs [at the beginning](https://github.com/rust-lang/triagebot/pull/1623/files#diff-d25ea18904fcfebcc9d004f49ffb091dd83ae65b4eaad2c0b01ad238b2210118R43) of the function, hope it's enough :)